src/shared/cpu-set-util.c

```diff
@@ -82,6 +82,53 @@ char *cpu_set_to_range_string(const CPUSet *set) {
         return TAKE_PTR(str) ?: strdup("");
 }
 
+char* cpu_set_to_mask_string(const CPUSet *a) {
+        _cleanup_free_ char *str = NULL;
+        size_t len = 0;
+        bool nz = 0;
+        int i, r;
+
+        assert(a);
+
+        /* Return CPU set in hexadecimal bitmap mask, e.g.
+         *   CPU   0 ->  "1"
+         *   CPU   1 ->  "2"
+         *   CPU 0,1 ->  "3"
+         *   CPU 0-3 ->  "f"
+         *   CPU 0-7 -> "ff"
+         *   CPU 4-7 -> "f0"
+         *   CPU   7 -> "80"
+         *   None    ->  "0" */
+
+        for (i = a->allocated * 8; i >= 0; i -= 4) {
+                unsigned char m = 0;
+
+                if (CPU_ISSET_S(i, a->allocated, a->set))
+                        m |= 1;
+                if (CPU_ISSET_S(i + 1, a->allocated, a->set))
+                        m |= 2;
+                if (CPU_ISSET_S(i + 2, a->allocated, a->set))
+                        m |= 4;
+                if (CPU_ISSET_S(i + 3, a->allocated, a->set))
+                        m |= 8;
+
+                if (!nz)
+                        nz = m > 0;
+
+                if (!nz && m == 0)
+                        continue;
```

Comment:
i don't get this?

Reply:
to skip leading zeros

Reply:
please add a comment about this.

Reply:
uh, I've already added `/* Skip leading zeros */`